### ISSUE TITLE
SA1013 conflicts with new C# pattern matching syntax

### ISSUE THREAD

**Issue by Szer** [DotNetAnalyzers/StyleCopAnalyzers#3222]
StyleCop throws false positive warnings for new C# destructing syntax:
```csharp
switch(ValidateCertificate(certificate)) {
   ...
   case AuthorizationResult { Authorized: true }:
      ...
}
[SA1013] Closing brace should be followed by a space.
```

Fixing doesn't help
```csharp
case AuthorizationResult { Authorized: true }>>SPACE HERE<<:
[SA1024] Colon should not be preceded by a space
```

The only workaround suppress it:
```
[SuppressMessage("StyleCop.CSharp.SpacingRules", "SA1013:ClosingBracesMustBeSpacedCorrectly",
            Justification = "New deconstructing pattern matching syntax conflicts with StyleCop rule")]
```

**Comment by sharwell** [comment]
Duplicate of #3053
